### DISCUSSION TITLE
Changing Theme

### DISCUSSION THREAD

**Discussion by Maikel1963** [RibirX/Ribir#610] [Q&A] [answered]
Hello, 
I am not content with the colors RIBIR is using as default. So I wanted to change the palette and started (just to get a feeling how it works) with:
```
let mut my_theme = FullTheme::default();
my_theme.palette.primary = Color::RED;
```
but I get the message that it is not possible to change something in theme like that.
How do I have to assign a diffenrent primary color to "my_theme"?

**Comment by M-Adoo** [comment, marked as answer]
I guess what you want is:

```rust
let mut palette = Palette::default();
palette.primary = Color::RED;
FullTheme {
  palette: Rc::new(palette),
  ..Default::default()
}
```


By the way, Ribir's theme is currently read-only. At runtime you can only replace the entire theme or do a partial reload via InheritTheme. However, the theme API is not yet stable and will change in the next one or two releases.
We may allow direct modification of themes at runtime.

**Reply by Maikel1963** [reply]
Fine! This does work. 
I do not need to alter the palette at runtime. I just want to apply my companies scheming for all coming projects.
So, how do I provide a FullTheme adhering to my companies standards without changing all values manually?

**Reply by M-Adoo** [reply]
I plan to provide a tool to convert css to themes in the future, but this is not in the short term plans. Right now it seems you need to do it all manually.

**Reply by Maikel1963** [reply]
Ok, I think I can build a lib providing a FullTheme after changing everything to my needs, so that it is the same in all applications. That's enough for me now. But I guess building it with your tool will make the  executable smaller.
